src/os/KeyValueDB.cc

```diff
@@ -11,21 +11,23 @@
 #endif
 
 KeyValueDB *KeyValueDB::create(CephContext *cct, const string& type,
-			       const string& dir)
+			       const string& dir, const bool default_sync_transaction)
```

Comment:
Hmm, I wonder if it is simpler to just make sure all of the callers are correctly using either submit_transaction or submit_transaction-sync?  For example, MonitorDBStore is using submit_transaction_sync(), which is correct.  DBObjectMap uses submit_transaction(), which is okay since FileStore is doing syncfs().